research/object_detection/core/region_similarity_calculator.py

```diff
@@ -66,18 +72,49 @@ class IouSimilarity(RegionSimilarityCalculator):
   This class computes pairwise similarity between two BoxLists based on IOU.
   """
 
-  def _compare(self, boxlist1, boxlist2):
+  def _compare(self, boxlist1, boxlist2,
+               groundtruth_labels=None, predicted_labels=None):
     """Compute pairwise IOU similarity between the two BoxLists.
 
     Args:
       boxlist1: BoxList holding N boxes.
       boxlist2: BoxList holding M boxes.
+      groundtruth_labels: a Tensor of shape [num_boxes, num_classes]
+        containing groundtruth labels.
+      predicted_labels: a Tensor of shape [num_boxes, num_classes]
+        containing predicted labels.
 
     Returns:
       A tensor with shape [N, M] representing pairwise iou scores.
     """
     return box_list_ops.iou(boxlist1, boxlist2)
 
+class DETRSimilarity(RegionSimilarityCalculator):
+  """Class to compute similarity for the Detection Transformer model.
+
+  This class computes pairwise similarity between two BoxLists using a weighted
+  combination of IOU, classification scores, and the L1 loss.
+  """
+
+  def _compare(self, boxlist1, boxlist2,
+               groundtruth_labels=None, predicted_labels=None):
+    """Compute pairwise IOU similarity between the two BoxLists.
+
+    Args:
+      boxlist1: BoxList holding N boxes.
+      boxlist2: BoxList holding M boxes.
+      groundtruth_labels: a Tensor of shape [num_boxes, num_classes]
+        containing groundtruth labels.
+      predicted_labels: a Tensor of shape [num_boxes, num_classes]
+        containing predicted labels.
+
+    Returns:
+      A tensor with shape [N, M] representing pairwise iou scores.
+    """
+    classification_scores = tf.matmul(groundtruth_labels,
+        tf.nn.softmax(predicted_labels), transpose_b=True)
+    return -5 * box_list_ops.l1(boxlist1, boxlist2) + 2 * box_list_ops.giou(
```

Comment:
add the scale factors 5 and 2 as constructor arguments. Also are these values mentioned in the paper ?